### DISCUSSION TITLE
Support inherited classes with reflection

### DISCUSSION THREAD

**Discussion by decades42** [stephenberry/glaze#586] [Q&A] [answered]
First of all, thank you for this awesome library, I've been trying to migrate my old projects with this library and most of them work flawlessly. However the library doesn't support inherited classes with reflection on, although I can disable reflection on them and use macros instead, it would be tedious to type when there are too many fields in the class

**Comment by stephenberry** [comment, marked as answer]
Right now reflection only works on [aggregate initializable](https://en.cppreference.com/w/cpp/language/aggregate_initialization) structs, which rules out inheritance. This is a limitation you will see for any library currently implementing pure reflection, until we get C++26 reflection. There may be a work around this limitation, but currently I would recommend using macros or `glz::meta`. Thankfully the `glz::meta` definitions will now reflect on your member pointer names, so that saves typing.

**Reply by decades42** [reply]
Thank you for the reply, I think it would be better to clarify that the current status of the library doesn't support inherited classes in readme or pure-reflection document :)

**Reply by schaumb** [reply]
Hi @stephenberry !
This isn't true (only at boost pfr, because I didn't have time to make it work with the empty structs with multiple inheritances ). 
You can implement it with the same technique that I did [here](https://github.com/vizzuhq/vizzu-lib/blob/main/src/base/refl/auto_struct.h#L29).

**Reply by stephenberry** [reply]
I understand that you may be able to get names from inherited non-aggregate structs, but can you use this loophole to get pointers?

**Reply by schaumb** [reply]
Yes of course. The loophole technique is way more powerful than structure binding. Currently, it is only used for aggregate initializable [member type-getters](https://github.com/boostorg/pfr/blob/develop/include/boost/pfr/detail/core14_loophole.hpp) (and information leaking hacks like [this](https://github.com/martong/access_private)). 
There are unused features like [recognize template-constructor](https://github.com/boostorg/pfr/issues/145) or [get list of the aggregates base classes](https://github.com/vizzuhq/vizzu-lib/blob/main/src/base/refl/auto_struct.h#L234-L236) 
After that, you can select which were the base classes, and you can cast the main object to it.

**Reply by scalpel4k** [reply]
is this something that could make it back into the open issue list?

**Reply by stephenberry** [reply]
@scalpel4k, I'm open to pull requests that add inheritance support, but I'm a bit leery of implementing it in C++20. Reflection is coming in C++26 and will make any code written to support this obsolete. There are a lot of higher priorities for the library right now, which include handling streams, networking, modules, and better runtime/compile-time performance.

**Comment by 134579** [comment]
Since we already support C++26 reflections, can we make Glaze support reflection on base class members as well? It seems that currently Glaze can only reflect members of the derived class.
https://gcc.godbolt.org/z/YWez6q7Yb
